### PR TITLE
Feature/homework 07

### PR DESCRIPTION
**Pull request by igoryushchenko**

### CODE REVIEW

File: homework-04/controllers/authMongoController.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import jwt from 'jsonwebtoken'
```
Comment (AlexeyBarkow) [inline homework-04/controllers/authMongoController.js]: DRY: try to reuse as much as you can. If you had separated separate all DB-specific functions to a separate file, the process of refactoring you code to use the new DB would have required much less code

File: homework-04/controllers/authMongoController.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import User from '../models/mongo/User'
+
+function localAuth (req, res) {
```
Comment (AlexeyBarkow) [inline homework-04/controllers/authMongoController.js]: all auth logic should stay where it was staying before before

Comment (AlexeyBarkow) [inline homework-04/controllers/authMongoController.js]: This controller func should take the following arguments:
function loalAuth(String email, String password): Promise<DBData>

File: homework-04/controllers/citiesMongoController.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import { City } from '../models/mongo/City'
+
+function getRandomCity (req, res) {
```
Comment (AlexeyBarkow) [inline homework-04/controllers/citiesMongoController.js]: the same for all other routes

File: homework-04/controllers/citiesMongoController.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    } else {
+      res.json(data)
+    }
+  })
+}
+
+function getAllCities (req, res) {
+  City.find((err, cities) => {
+    if (err) {
+      res.status(500).json({
```
Comment (AlexeyBarkow) [inline homework-04/controllers/citiesMongoController.js]: all res / req logic should stay inside router/ folder

File: homework-04/populatedb.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import { City, Location } from './models/mongo/City'
```
Comment (AlexeyBarkow) [inline homework-04/populatedb.js:2]: For me this doesn't work. did you forget to connect to database?

**Review by AlexeyBarkow — CHANGES_REQUESTED** [review]

File: homework-04/controllers/authMongoController.js (hunk 1 above)
Comment (igoryushchenko) [reply homework-04/controllers/authMongoController.js]: Fixed

File: homework-04/controllers/authMongoController.js (hunk 2 above)
Comment (igoryushchenko) [reply homework-04/controllers/authMongoController.js]: Fixed

Comment (igoryushchenko) [reply homework-04/controllers/authMongoController.js]: fixed

File: homework-04/controllers/citiesMongoController.js (hunk 3 above)
Comment (igoryushchenko) [reply homework-04/controllers/citiesMongoController.js]: Fixed for all controllers

File: homework-04/populatedb.js (hunk 5 above)
Comment (igoryushchenko) [reply homework-04/populatedb.js:2]: Connection has been added

File: homework-04/controllers/citiesMongoController.js (hunk 4 above)
Comment (igoryushchenko) [reply homework-04/controllers/citiesMongoController.js]: Fixed.

File: homework-04/app.js [hunk 6]
```diff
@@ -1,35 +1,42 @@
...
 app.use(express.urlencoded())
 apiRouter.use('/auth', authentication)
-apiRouter.use('/users', users)
 
+apiRouter.use('/users', users)
 apiRouter.use('/products', products)
+apiRouter.use('/cities', cities)
 app.all('*', cookieParser, queryParser)
 app.all(/^\/api\/(?!auth*).*$/, verifyToken)
 app.use('/api', apiRouter)
```
Comment (AlexeyBarkow) [inline homework-04/app.js:41]: I'd rather use bodyparser middleware for the whole /api router than set in in each route

File: homework-04/models/mongo/lastModifiedPlugin.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+function lastModifiedPlugin (schema, options) {
+  schema.add({ lastModifiedDate: Date })
+
+  schema.pre('save', function (next) {
```
Comment (AlexeyBarkow) [inline homework-04/models/mongo/lastModifiedPlugin.js:4]: Please, do the same thing on schema update

**Review by AlexeyBarkow — CHANGES_REQUESTED** [review]

File: homework-04/models/mongo/lastModifiedPlugin.js (hunk 7 above)
Comment (igoryushchenko) [reply homework-04/models/mongo/lastModifiedPlugin.js:4]: Done

File: homework-04/app.js (hunk 6 above)
Comment (igoryushchenko) [reply homework-04/app.js:41]: Fixed

**Review by AlexeyBarkow — APPROVED** [review]

**Pull request merged**